ext/ddtrace.c

```diff
@@ -2082,11 +2076,28 @@ static PHP_FUNCTION(consume_distributed_tracing_headers) {
 
 /* {{{ proto array generate_distributed_tracing_headers() */
 static PHP_FUNCTION(generate_distributed_tracing_headers) {
-    UNUSED(execute_data);
+    zend_array *inject = NULL;
+
+    ZEND_PARSE_PARAMETERS_START(0, 1)
+        Z_PARAM_OPTIONAL
+        Z_PARAM_ARRAY_HT(inject)
+    ZEND_PARSE_PARAMETERS_END();
 
     array_init(return_value);
     if (get_DD_TRACE_ENABLED()) {
-        ddtrace_inject_distributed_headers(Z_ARR_P(return_value), true);
+        if (inject) {
+            zend_array *inject_set = zend_new_array(zend_hash_num_elements(inject));
+            zval *val;
+            ZEND_HASH_FOREACH_VAL(inject, val) {
+                if (Z_TYPE_P(val) == IS_STRING) {
+                    zend_hash_add_empty_element(inject_set, Z_STR_P(val));
+                }
+            } ZEND_HASH_FOREACH_END();
```

Comment:
Question: this technically filters out all non-`string` values from the array passed as first argument to this function, by only copying `string` values to the `inject_set` hashmap, correct?
In case a value is non-`string`, we end up having less items in this hashmap then we initialised it with in line 2089:
```c
zend_array *inject_set = zend_new_array(zend_hash_num_elements(inject));
```
IDK if this might be a problem (like having empty array elements in userland at the end of the array), just making sure I get it right.

Reply:
That's right, but no, the initialization size is just about how many entries to emalloc() in hashtable init, essentially it just saves you doing reallocs while filling.